src/spv/inst-read.cpp

```diff
@@ -363,6 +389,31 @@ Instruction* Instruction::readOp(
         handle_type(type, inst.operands, words, i);
     }
 
+    // If it was an extension, make sure it's supported
+    // TODO: should differentiate "extensions" and "extended instructions"?
+    if (op == spv::OpExtension) {
+        auto& operands = inst.operands;
+        assert(operands[0].type == Token::Type::STRING);
+        std::string ext_name = std::get<std::string>(operands[0].raw);
+
+        // TODO: maybe change to "unordered_set" if there's a lot of supported extensions
+        // TODO: move into "Spv::Instruction" class?
+        // Contains only implemented extensions.
+        const std::vector<std::string> supported_ext {
```

Comment:
I agree with your TODOs. This seems to fit better by the enum class Extension in instruction.cxx. You could even make a static isExtensionPresent method or something, since I suspect we will use this behavior again.

Reply:
Do you think extensions and extended instructions should be separate? Currently, the interpreter considers `GLSL.std.` an extension rather than an extended instruction, though I'm not sure about the differences yet.

Reply:
The main difference between extensions and extended instructions is whether they are used like Capabilities (for the former) or a generic instruction extension. That is a fairly superficial distinction.

I have thought about this a lot and now I am on the fence whether we want to check extensions at all. We don't check Capabilities, for example, so it makes sense we wouldn't check extensions if they are not directly useful.

I made a design choice earlier not to worry too much about decorations and capabilities. There are many, they aren't always relevant, and SPIR-V continues to change, so it makes the interpreter more robust if it can ignore decorations it doesn't need. On the other hand, that opens the door for silent failures.

Reply:
Please remove this logic. I see that examples/opencl-simple is failing with error "Unsupported extension: SPV_KHR_storage_buffer_storage_class".